CI: Add workflow_dispatch to workflows that still lack it

kinkie opened the pull request:
Manually triggering a workflow rerun is handy when troubleshooting. Our
coverity-scan.yaml workflow already has a workflow_dispatch trigger.

kinkie commented:
@rousskov I suspect that right now our queue is stuck because all eligible PRs are queued to run the checks defined in quick.yaml, however that workflow had been disabled (I don't know by whom or why), so the trigger events passed by without engaging it.
The workflow is meant to be triggered on push or PR, but if I were to push, approvals would be lost and would have to be re-granted. This PR would allow manually running these workflows if we ever were to find ourselves again in this situation, as well as allowing to un-stick it

rousskov reviewed (dismissed):
> @rousskov I suspect that right now our queue is stuck because all eligible PRs are queued to run the checks defined in quick.yaml, however that workflow had been disabled (I don't know by whom or why)

I have not changed any configuration settings since you have started working on fixing CodeQL (and I would have disclosed those changes if I were to do them). I do not see any signs[^1] of a disabled workload in this repository right now (and this PR tests did run). Did you re-enable it?

If at all possible, please start sharing any changes to repository configuration, especially if they were not previously discussed. CC: @yadij.

[^1]: FWIW, you can see how a disabled workload is reported at https://github.com/measurement-factory/squid/actions

> so the trigger events passed by without engaging it. The workflow is meant to be triggered on push or PR, but if I were to push, approvals would be lost and would have to be re-granted. This PR would allow manually running these workflows if we ever were to find ourselves again in this situation, as well as allowing to un-stick it

As you know from recent [exchanges](https://github.com/squid-cache/squid/pull/1958/files/7a8009eea24616e7e7838830d4583731e0afd7f0#r1875022234), I fought for adding this useful trigger. Thank you for adding it to the remaining workloads!

I have adjusted PR title/description to remove the false implication that all workloads lacked this trigger. I also removed a statement regarding permissions required to use this trigger because, [AFAICT](https://github.com/orgs/community/discussions/26622#discussioncomment-3252561), it can be used by non-owners in certain cases (and that is OK!).

kinkie commented:
> > @rousskov I suspect that right now our queue is stuck because all eligible PRs are queued to run the checks defined in quick.yaml, however that workflow had been disabled (I don't know by whom or why)
> 
> I have not changed any configuration settings since you have started working on fixing CodeQL (and I would have disclosed those changes if I were to do them). I do not see any signs[1](#user-content-fn-1-2f12e055b4d89248eaf6cbd60c2c0c61) of a disabled workload in this repository right now (and this PR tests did run). Did you re-enable it?

Yes, I did.

> If at all possible, please start sharing any changes to repository configuration, especially if they were not previously discussed. CC: @yadij.

Sure.
So far I have just disabled CodeQL checks, after trying pointlessly to reconfigure them.

> > so the trigger events passed by without engaging it. The workflow is meant to be triggered on push or PR, but if I were to push, approvals would be lost and would have to be re-granted. This PR would allow manually running these workflows if we ever were to find ourselves again in this situation, as well as allowing to un-stick it
> 
> As you know from recent [exchanges](https://github.com/squid-cache/squid/pull/1958/files/7a8009eea24616e7e7838830d4583731e0afd7f0#r1875022234), I fought for adding this useful trigger. Thank you for adding it to the remaining workloads!

No worries. In fact, in hindsight, I am surprised this is not on by default


> I have adjusted PR title/description to remove the false implication that all workloads lacked this trigger. I also removed a statement regarding permissions required to use this trigger because, [AFAICT](https://github.com/orgs/community/discussions/26622#discussioncomment-3252561), it can be used by non-owners in certain cases (and that is OK!).

Thanks

> 
> ## Footnotes
> 1. FWIW, you can see how a disabled workload is reported at https://github.com/measurement-factory/squid/actions [↩](#user-content-fnref-1-2f12e055b4d89248eaf6cbd60c2c0c61)

rousskov commented:
> The workflow is meant to be triggered on push or PR, but if I were to push, approvals would be lost

It does not really matter in _this_ case (we have enough folks to quickly re-approve those PRs), but pushing an _empty_ PR commit does not invalidate past PR approvals. GitHub apparently added that behavior in [2023](https://github.com/orgs/community/discussions/41547), and I have just successfully tested that with #1966 (commit 4edfc206).

rousskov commented:
> kinkie added[ M-cleared-for-merge ](https://github.com/squid-cache/squid/labels/M-cleared-for-merge)S-could-use-an-approval and removed[ S-waiting-for-author ](https://github.com/squid-cache/squid/labels/S-waiting-for-author) labels

@kinkie, would you mind addressing my change requests before merging?
